[17.0][MIG]stock_no_negative: Migrated to v17

fd-oerp opened the pull request.

alexis-via committed:
[MIG] stock_no_negative: Migration to 10.0

serpentcs-dev1 committed:
[ADD] test cases

MiquelRForgeFlow committed:
[IMP] stock_no_negative: Behave correctly on other tests

etobella committed:
[MIG] stock_no_negative: Migration to 11.0

rgarnau committed:
minor fixes

etobella committed:
[FIX] Issue #391

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] stock_no_negative: Incorrect tests

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update stock_no_negative.pot

osoul-dev committed:
Added translation using Weblate (Arabic)

osoul-dev committed:
Translated using Weblate (Arabic)

Currently translated at 50.0% (4 of 8 strings)

Translation: stock-logistics-workflow-11.0/stock-logistics-workflow-11.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-11-0/stock-logistics-workflow-11-0-stock_no_negative/ar/

pedrobaeza committed:
[IMP] stock_no_negative: Avoid problems on integration tests

For example, when you install *delivery* along with this one

JordiBForgeFlow committed:
[stock_no_negative][IMP] Add the ability to allow negative stock for individual
stock locations.

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update stock_no_negative.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

JordiBForgeFlow committed:
fix in incorrect view

meigallodixital committed:
[MIG] stock_no_negative: Migration to 12.0

oca-travis committed:
[UPD] Update stock_no_negative.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_no_negative/

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

qdelcourte committed:
[IMP] stock_no_negative: black, isort

qdelcourte committed:
[MIG] stock_no_negative: Migration to 13.0

qdelcourte committed:
[UPD] CONTRIBUTORS.rst and HISTORY.rst

Commit:
[12.0][FIX] stock_no_negative: Use product display_name instead of name

oca-travis committed:
[UPD] Update stock_no_negative.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-13.0/stock-logistics-workflow-13.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-13-0/stock-logistics-workflow-13-0-stock_no_negative/

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

OCA-git-bot committed:
[UPD] README.rst

Vanneri committed:
[IMP] stock_no_negative: black, isort, prettier

Vanneri committed:
[MIG] stock_no_negative: Migration to 14.0

oca-travis committed:
[UPD] Update stock_no_negative.pot

OCA-git-bot committed:
[UPD] README.rst

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 84.6% (11 of 13 strings)

Translation: stock-logistics-workflow-14.0/stock-logistics-workflow-14.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-14-0/stock-logistics-workflow-14-0-stock_no_negative/pt/

dreispt committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (13 of 13 strings)

Translation: stock-logistics-workflow-14.0/stock-logistics-workflow-14.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-14-0/stock-logistics-workflow-14-0-stock_no_negative/pt/

OCA-git-bot committed:
stock_no_negative 14.0.1.0.1

JoanMForgeFlow committed:
[IMP][15.0] stock_no_negative: black, isort, prettier

JoanMForgeFlow committed:
[15.0][MIG] stock_no_negative: Migration to 15.0

Commit:
[UPD] Update stock_no_negative.pot

OCA-git-bot committed:
[UPD] README.rst

luffah committed:
Translated using Weblate (French)

Currently translated at 100.0% (10 of 10 strings)

Translation: stock-logistics-workflow-15.0/stock-logistics-workflow-15.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-15-0/stock-logistics-workflow-15-0-stock_no_negative/fr/

vvanneri-ia committed:
[MIG] stock_no_negative: Migration to 16.0

Commit:
[UPD] Update stock_no_negative.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_no_negative 16.0.1.0.1

yostashiro committed:
[IMP] stock_no_negative: add option to skip negative qty check

badbole committed:
Added translation using Weblate (Croatian)

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_no_negative 16.0.1.0.2

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_no_negative/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: stock-logistics-workflow-16.0/stock-logistics-workflow-16.0-stock_no_negative
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-16-0/stock-logistics-workflow-16-0-stock_no_negative/es/

LuisBrose committed:
Added translation using Weblate (German)

mymage committed:
Added translation using Weblate (Italian)

rousseldenis commented on stock_no_negative/__manifest__.py:
```diff
@@ -1,15 +1,15 @@
-# ?? 2015-2016 Akretion (http://www.akretion.com)
-# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# ?? 2023-2024 OERP Canada (https://www.oerp.ca/)
```
You cannot change authorship

rousseldenis commented on stock_no_negative/__manifest__.py:
```diff
@@ -1,15 +1,15 @@
     "license": "AGPL-3",
     "summary": "Disallow negative stock levels by default",
-    "author": "Akretion,Odoo Community Association (OCA)",
+    "author": "OERP Canada,Odoo Community Association (OCA)",
```
Same here.

rousseldenis commented on stock_no_negative/i18n/fr.po:
```diff
@@ -73,7 +68,7 @@ msgstr ""
 #: model:ir.model,name:stock_no_negative.model_stock_location
 msgid "Inventory Locations"
-msgstr "Emplacements d'inventaire"
+msgstr "Emplacements d'inventair"
```
Ah oui ?

rousseldenis commented on stock_no_negative/i18n/fr.po:
```diff
@@ -96,24 +91,10 @@ msgstr "Quants"
-#~ "produit %s'%s deviendrait négative(%s) dans l'emplacement du stock '%s'. "
-#~ "Un stock négatif n'est pas permis pour ce produit."
+"Impossible de valider cette opération car le niveau de stock de ce "
+"produit %s'%s deviendrait négative(%s) dans l'emplacement du stock '%s'. "
```
le niveau de stock... négati**f**

fd-oerp replied:
Hi @rousseldenis is there anything wrong with this translation ? Do I need to keep it as it is the way it's commented in v16 ?

rousseldenis replied:
Why have you changed this string ?

fd-oerp replied:
> Why have you changed this string ?

Hi @rousseldenis, haven't changed those strings , I just made those 3 lines of translation uncommented , should I keep it commented ?

rousseldenis replied:
Ok, didn't see. Could you replace 'négative' by 'négatif' ? Thanks

fd-oerp replied:
Hi @rousseldenis , changes are updated with this commit : [1c192eb62d90a680514c695a2f4090a614e0fecd](https://github.com/fd-oerp/stock-logistics-workflow/commit/1c192eb62d90a680514c695a2f4090a614e0fecd)

rousseldenis reviewed (changes requested):
@fd-oerp Thanks for this. Some changes

rousseldenis commented:
/ocabot migration stock_no_negative

fd-oerp commented:
Hi @rousseldenis , Pre-commit fails are updated with this commit : [1c507c46f9cedc621ad903b864b5efcec3893fa6](https://github.com/fd-oerp/stock-logistics-workflow/commit/1c507c46f9cedc621ad903b864b5efcec3893fa6)

fd-oerp commented:
Hi @rousseldenis , Is there anything needs to be done from my side or do I need to squash the commits ?

celm1990 reviewed (approved):
Please squash commits administratives(Weblate)

fd-oerp commented:
> Please squash commits administratives(Weblate)

Hi @celm1990 , Do I need to do anything with Weblate while squashing or just need to squash the PR commits ?

celm1990 commented:
> > Please squash commits administratives(Weblate)
> 
> Hi @celm1990 , Do I need to do anything with Weblate while squashing or just need to squash the PR commits ?

according this documentation please, https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate


For example, latest commits must be squashed similar to
![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/928709a7-14e0-4d0c-babc-5ea55817650f)

fd-oerp commented:
> > > Please squash commits administratives(Weblate)
> > 
> > 
> > Hi @celm1990 , Do I need to do anything with Weblate while squashing or just need to squash the PR commits ?
> 
> according this documentation please, https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate
> 
> For example, latest commits must be squashed similar to ![image](https://private-user-images.githubusercontent.com/7775116/289687503-928709a7-14e0-4d0c-babc-5ea55817650f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzOTQ5MjEsIm5iZiI6MTcwMjM5NDYyMSwicGF0aCI6Ii83Nzc1MTE2LzI4OTY4NzUwMy05Mjg3MDlhNy0xNGUwLTRkMGMtYmFiYy01ZWE1NTgxNzY1MGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTJUMTUyMzQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzQ5ZDFmMmFjNWM1NmQwM2I5MDJjMTEwNjFhNDAyNGFlZjViMzk1OGY0ZDUwMjE2NTJmYTY1OWJhMDRiMjQwNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.JoOdHz0wvC_hxdRXbsMUHXKClZNYp2hR9yih1aUkOkw)

Hi @celm1990 I squashed the latest commits/updated the PR by latest commits in the head repo.

celm1990 commented:
Sorry but i not see squash commits, as per the guidelines, I recommend squashing these commits for better clarity and adherence to the standards(Same language, same author, so can be squash)
![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/e13be6e6-6329-47f9-a8c4-476203f2f60c)


![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/515aa460-4646-48d1-a606-e3efdffe619b)

![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/27544522-bce4-4e42-afc9-7110f537a4fc)

rousseldenis commented:
@fd-oerp Moreover you did a merge of 17.0 branch  in yours instead of rebase. Could you remove that merge commit ?

rousseldenis commented:
@fd-oerp Is it normal that you try to do a PR from your local 17.0 branch ? You should have a specific branch IMHO

fd-oerp commented:
> Sorry but i not see squash commits, as per the guidelines, I recommend squashing these commits for better clarity and adherence to the standards(Same language, same author, so can be squash) ![image](https://private-user-images.githubusercontent.com/7775116/290013479-e13be6e6-6329-47f9-a8c4-476203f2f60c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0ODU0OTYsIm5iZiI6MTcwMjQ4NTE5NiwicGF0aCI6Ii83Nzc1MTE2LzI5MDAxMzQ3OS1lMTNiZTZlNi02MzI5LTQ3ZjktYThjNC00NzYyMDNmMmY2MGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTNUMTYzMzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjM5Y2ZkMzU4OThlYmUyMzVhZDU4ODg2YWNiNTYxZDE5NmJlNzlhOTEzZGRhYjJkNGYyODJkNjQ2MjlmOWI5OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.WvaZNVmj6SfiUUsUtm7zvHe8tw7c9zQMzUqYqRWv6hg)
> 
> ![image](https://private-user-images.githubusercontent.com/7775116/290013059-515aa460-4646-48d1-a606-e3efdffe619b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0ODU0OTYsIm5iZiI6MTcwMjQ4NTE5NiwicGF0aCI6Ii83Nzc1MTE2LzI5MDAxMzA1OS01MTVhYTQ2MC00NjQ2LTQ4ZDEtYTYwNi1lM2VmZGZmZTYxOWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTNUMTYzMzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzUzMzMwODczZDdhN2I3Nzk3NjQzNGM1OGVmNjQ1MDM5Y2QxMTcxMzM4ZjdiMjVkYzJkMjNhNjhlNTg1YzIxNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.jQfQ4CClkvoOmwdFZkcWRtk99i3mN2PBTi2Z2iODeXc)
> 
> ![image](https://private-user-images.githubusercontent.com/7775116/290013231-27544522-bce4-4e42-afc9-7110f537a4fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI0ODU0OTYsIm5iZiI6MTcwMjQ4NTE5NiwicGF0aCI6Ii83Nzc1MTE2LzI5MDAxMzIzMS0yNzU0NDUyMi1iY2U0LTRlNDItYWZjOS03MTEwZjUzN2E0ZmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTNUMTYzMzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGUyZTBkYTE2MjExNGQ5M2E0MjBlNDRmMjk2MDI3ZWUyMzE4ZWJlMDJiNWM4Y2M5ZDVjNWY3N2Y5YmYzMTZkZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.m7i91ipFiq4-M2MZcZBInvgY6Lu8cy9TDYPJ8Dniwuk)

Hi @celm1990 , My bad I haven't got you correctly , now I have squashed the commits from the same author:
![git1](https://github.com/OCA/stock-logistics-workflow/assets/150078409/3f0ef7d0-1970-4b68-aae2-1e37d547141e)
Let me know if still anything needs to be done from my side, Thanks.

fd-oerp commented:
> @fd-oerp Moreover you did a merge of 17.0 branch in yours instead of rebase. Could you remove that merge commit ?

Hi @rousseldenis My bad, I have removed that merge commit.

fd-oerp commented:
> @fd-oerp Is it normal that you try to do a PR from your local 17.0 branch ? You should have a specific branch IMHO

@rousseldenis okay, How should I proceed further , should I delete this PR and make a new PR by creating a specific branch ?

rousseldenis commented:
> > @fd-oerp Is it normal that you try to do a PR from your local 17.0 branch ? You should have a specific branch IMHO
> 
> @rousseldenis okay, How should I proceed further , should I delete this PR and make a new PR by creating a specific branch ?

Not necessary, just wanted to be sure it's ok on your side

fd-oerp commented:
> > > @fd-oerp Is it normal that you try to do a PR from your local 17.0 branch ? You should have a specific branch IMHO
> > 
> > 
> > @rousseldenis okay, How should I proceed further , should I delete this PR and make a new PR by creating a specific branch ?
> 
> Not necessary, just wanted to be sure it's ok on your side

Yes @rousseldenis , It's okay from my side

smcqueen81 reviewed (approved):
Funcional OK

fd-oerp commented:
Hi @rousseldenis is there anything I need to do from my side in this PR ?

rousseldenis commented on stock_no_negative/tests/test_stock_no_negative.py:
```diff
@@ -165,17 +165,27 @@ def test_true_allow_negative_stock_product_with_lot(self):
             ]
         )
-        self.assertEqual(quant.quantity, -100)
+        self.assertEqual(quant.quantity, 100)
```
@fd-oerp Why changing this as this test is checking negative is allowed ?

fd-oerp replied:
Hi @rousseldenis , the assert giving the positive quantity, I have also updated some code in this method

rousseldenis replied:
Be careful. During migration, the results of test should - generally - be kept.

As the test is reflecting the purpose of the code. Reading the comment at the beginning of the test function would have oriented you through a questioning.

So two questions remain:

- Is this module do not behave as expected?
- Has Odoo changed core module to avoid negative quantities?

Both should be answered.

fd-oerp replied:
Hi @rousseldenis ,
Thanks for clarity, 1) Yes this module works as expected 2)No, by using this module we can allow to have a negative quantity for product with lot as well.

celm1990 replied:
> Hi @celm1990 @rousseldenis any update on this PR, anything needs to be updated from my side?

It is not clear to me because you changed the tests, same doubt as @rousseldenis.

fd-oerp replied:
Hi @celm1990 

I just updated that test case with the necessary changes, please review and proceed or let me know if anything needs to be done from my side

rousseldenis reviewed (changes requested):
See above

fd-oerp commented:
> See above

Hi @rousseldenis , any comments from your side regarding this ?

Wodran14 commented:
@rousseldenis Any update on this PR?

yay-sw commented:
Hi @rousseldenis , any updates regarding this?

fd-oerp commented:
Hi @celm1990 @rousseldenis any update on this PR, anything needs to be updated from my side?

celm1990 commented:
@fd-oerp 
Please split your commit according guideslines https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0
![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/3f1489c8-0e84-490d-ad00-8b05090727d6)

A first commit only wit `pre-commit run`, and second commit with all changes related to migration
![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/6b081acf-e9a2-4a72-aa23-a3098d40940f)

fd-oerp committed:
[IMP] stock_no_negative: pre-commit auto fixes

fd-oerp commented:
> @fd-oerp Please split your commit according guideslines https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0 ![image](https://private-user-images.githubusercontent.com/7775116/309561696-3f1489c8-0e84-490d-ad00-8b05090727d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1Njc0MzMsIm5iZiI6MTcwOTU2NzEzMywicGF0aCI6Ii83Nzc1MTE2LzMwOTU2MTY5Ni0zZjE0ODljOC0wZTg0LTQ5MGQtYWQwMC04YjA1MDkwNzI3ZDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDRUMTU0NTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGU0NzVhZjM1ZTNlZWJiNjc5MGM0ODIyYjMxNmY5YjM0NTJlOGZhZTNkMWFlNmJkN2QwMDVmZWUxY2E2MWIwZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FmoOVh0zsRvwLBldRnjrHupIt3dn60kKhWQVU-o-wLo)
> 
> A first commit only wit `pre-commit run`, and second commit with all changes related to migration ![image](https://private-user-images.githubusercontent.com/7775116/309561770-6b081acf-e9a2-4a72-aa23-a3098d40940f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1Njc0MzMsIm5iZiI6MTcwOTU2NzEzMywicGF0aCI6Ii83Nzc1MTE2LzMwOTU2MTc3MC02YjA4MWFjZi1lOWEyLTRhNzItYWEyMy1hMzA5OGQ0MDk0MGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDRUMTU0NTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTRiZWY0YjIzNmY3YWI1OTY1NWM2MDg3NjhlMzFjZjdlMjlkNGM5YThjNjk1YTM0NjIxNmI3NDdmNmEwZTUwYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.CtcSDIA-tWomXKW1xDtorEm76xdKody9PBAyXGTQXJs)

Hi @celm1990 splitted the commits as per the guidelines(precommit + mig), you can review and proceed, please.

celm1990 commented on stock_no_negative/tests/test_stock_no_negative.py:
```diff
@@ -166,16 +166,17 @@ def test_true_allow_negative_stock_product_with_lot(self):
             self.stock_picking_with_lot._action_done()
         self.stock_picking_with_lot.move_ids.move_line_ids[0].lot_id = self.lot1.id
         self.stock_picking_with_lot._action_done()
+        self.stock_picking_with_lot.button_validate()
```
Because you call  `button_validate` function after `_action_done` I suspect the test is failing due to another cause, and this is only hiding the problem.

celm1990 commented on stock_no_negative/tests/test_stock_no_negative.py:
```diff
@@ -166,16 +166,17 @@ def test_true_allow_negative_stock_product_with_lot(self):
             [
                 ("product_id", "=", self.product_with_lot.id),
                 ("location_id", "=", self.location_id.id),
-                ("lot_id", "=", self.lot1.id),
```
because this change, this alter the purpose of the test

fd-oerp replied:
> because this change, this alter the purpose of the test

Hi @celm1990, I have checked this same process in v17, 
In v16:
If I validate 5 qty in delivery 2 records will be generated in stock.quant, one with 5 other with -5 both have the lot number.

In v17:
If I validate 5 qty in delivery 2 records will be generated in stock.quant, one with 5 other with -5 but from these two only +5 one have the lot number.

fd-oerp replied:
Hi @celm1990 , I updated the test case by looking into it more deeply, Detailed operation(move line) with lot should be created before the done qty updated in move if we want to negative stock with lot no.
So I updated as per that.
Can you please review this!

celm1990 replied:
According to the test, the expected behavior is as follows:

1) Try to validate picking without a lot, and it should raise a `UserError`.
2) Assign a lot and validate the picking; now the picking should be done without error.

However, according to your changes, step 1 is missing, Why?.

fd-oerp replied:
Hi @celm1990 ,

I got your point, before I removed that because I created the move line(with lot) while the picking was created so the Usererror won't raise that's why I removed it. 

But I just updated that test case, and added the test case for that Usererror to be raised, and then I'm creating the Detaile operations (move line with lot).

Can you please review this!

john-herholz-dt reviewed (approved):
Functional check, works as expected.

celm1990 reviewed (approved):
LGTM

@fd-oerp please adapt your commit message
![image](https://github.com/OCA/stock-logistics-workflow/assets/7775116/8de1c403-cb3c-4c0f-8982-9e0f69ff565e)

fd-oerp commented:
> LGTM
> 
> @fd-oerp please adapt your commit message ![image](https://private-user-images.githubusercontent.com/7775116/310567083-8de1c403-cb3c-4c0f-8982-9e0f69ff565e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NDExODAsIm5iZiI6MTcwOTc0MDg4MCwicGF0aCI6Ii83Nzc1MTE2LzMxMDU2NzA4My04ZGUxYzQwMy1jYjNjLTRjMGYtODk4Mi05ZTBmNjlmZjU2NWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDZUMTYwMTIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWMxN2IyZjlmNWE0MWQ4NDg2YTY4MDI4YjNhNGU4ZmMzNjZmYjkzMjgyZmExZTdiZjM0NWFhNjJjZDAwMDNlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-CoRIia1SYasPgvkYsszQbAkiviyO8zbOYYUovwxIhs)

Hi @celm1990 , should I modify messages to just like 'migrated module..'  or something?
because these messages indicate what I have done, please lead me.

fd-oerp commented:
> > LGTM
> > @fd-oerp please adapt your commit message ![image](https://private-user-images.githubusercontent.com/7775116/310567083-8de1c403-cb3c-4c0f-8982-9e0f69ff565e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NDExODAsIm5iZiI6MTcwOTc0MDg4MCwicGF0aCI6Ii83Nzc1MTE2LzMxMDU2NzA4My04ZGUxYzQwMy1jYjNjLTRjMGYtODk4Mi05ZTBmNjlmZjU2NWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDZUMTYwMTIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWMxN2IyZjlmNWE0MWQ4NDg2YTY4MDI4YjNhNGU4ZmMzNjZmYjkzMjgyZmExZTdiZjM0NWFhNjJjZDAwMDNlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-CoRIia1SYasPgvkYsszQbAkiviyO8zbOYYUovwxIhs)
> 
> Hi @celm1990 , should I modify messages to just like 'migrated module..' or something? because these messages indicate what I have done, please lead me.

@celm1990 , or would it be fine if I delete all those commits (MIG once) and just do one commit for MIG ?

fd-oerp committed:
[17.0][MIG]stock_no_negative: Migrated module to v17.0
[17.0][MIG]stock_no_negative: Migrated module to v17.0

[17.0][MIG]stock_no_negative: Migrated module to v17.0

fd-oerp commented:
Hi @celm1990 , I updated the migrated module with just one MIG commit. please review!

fd-oerp commented:
Hi @rousseldenis , could you please review this and proceed ?

fd-oerp commented:
Hi @rousseldenis , Could you please review this and let me know if any changes are needed or proceed?

celm1990 commented:
@OCA/logistics-maintainers could somebody help me with revision and merge please?

rousseldenis reviewed (approved):
Code review

rousseldenis commented:
/ocabot merge nobump
